File: packages/shared-libs/go/auth/jwt.go
// shared-libs/go/auth
// JWT validation, consent enforcement, step-up auth middleware

package auth

import (
	"context"
	"fmt"
)

type Claims struct {
	Sub              string   `json:"sub"`
	Iss              string   `json:"iss"`
	Aud              []string `json:"aud"`
	ConsentTiers     []string `json:"consent_tiers"`
	Scopes           map[string][]string `json:"scopes"`
	StepupVerified   bool     `json:"stepup_verified"`
	AttStatus        string   `json:"att_status"`
	ConsentFlags     int      `json:"consent_flags"`
	DeviceFingerprint string  `json:"device_fingerprint"`
}

// ValidateConsent checks if user has required consent tier
func ValidateConsent(ctx context.Context, claims *Claims, tier string) error {
	for _, t := range claims.ConsentTiers {
		if t == tier {
			return nil
		}
	}
	return fmt.Errorf("missing consent tier: %s", tier)
}

// ValidateScope checks if JWT has required scope
func ValidateScope(ctx context.Context, claims *Claims, service string, action string) error {
	if scopes, ok := claims.Scopes[service]; ok {
		for _, s := range scopes {
			if s == action {
				return nil
			}
		}
	}
	return fmt.Errorf("missing scope: %s:%s", service, action)
}

File: packages/shared-libs/python/auth/jwt.py
# shared-libs/python/auth
# JWT validation, consent enforcement, step-up auth middleware

import jwt
from typing import List, Dict

class Claims:
    def __init__(self, payload: dict):
        self.sub = payload.get("sub")
        self.iss = payload.get("iss")
        self.aud = payload.get("aud", [])
        self.consent_tiers = payload.get("consent_tiers", [])
        self.scopes = payload.get("scopes", {})
        self.stepup_verified = payload.get("stepup_verified", False)
        self.att_status = payload.get("att_status")
        self.consent_flags = payload.get("consent_flags", 0)

def validate_consent(claims: Claims, tier: str) -> bool:
    """Check if user has required consent tier"""
    return tier in claims.consent_tiers

def validate_scope(claims: Claims, service: str, action: str) -> bool:
    """Check if JWT has required scope"""
    if service not in claims.scopes:
        return False
    return action in claims.scopes[service]
